InnerEye/ML/lightning_models.py

```diff
@@ -301,13 +301,14 @@ def compute_and_log_metrics(self,
                 logits[:, i, ...], targets[:, i, ...], subject_ids)
             # compute metrics on valid masked tensors only
             if masked is not None:
-                _posteriors = self.logits_to_posterior(masked.model_outputs.data)
+                _logits = masked.model_outputs.data
+                _posteriors = self.logits_to_posterior(_logits)
                 # Image encoders already prepare images in float16, but the labels are not yet in that dtype
                 _labels = masked.labels.data.to(dtype=_posteriors.dtype)
                 _subject_ids = masked.subject_ids
                 assert _subject_ids is not None
                 for metric in metric_list:
-                    metric(_posteriors, _labels)
+                    metric(_logits, _labels) if (isinstance(metric, ScalarMetricsBase) and metric.compute_from_logits is True) else metric(_posteriors, _labels)
```

Comment:
I think this expression would be easier on the eye if you did a plain vanilla if/else. also, "is True" is obsolete.
if isinstance(metrics, ...) and metric.compute_from_logits:
   metric(logits,labels)
else:
...